contracts/src/lib.rs

```diff
@@ -1,16 +1,93 @@
 #[cfg(feature = "bin")]
 pub mod paths;
 
+include!(concat!(env!("OUT_DIR"), "/BalancerV2Vault.rs"));
 include!(concat!(env!("OUT_DIR"), "/ERC20.rs"));
 include!(concat!(env!("OUT_DIR"), "/ERC20Mintable.rs"));
-include!(concat!(env!("OUT_DIR"), "/IUniswapLikeRouter.rs"));
+include!(concat!(env!("OUT_DIR"), "/GPv2AllowListAuthentication.rs"));
+include!(concat!(env!("OUT_DIR"), "/GPv2Settlement.rs"));
 include!(concat!(env!("OUT_DIR"), "/IUniswapLikePair.rs"));
-include!(concat!(env!("OUT_DIR"), "/UniswapV2Pair.rs"));
-include!(concat!(env!("OUT_DIR"), "/UniswapV2Router02.rs"));
-include!(concat!(env!("OUT_DIR"), "/UniswapV2Factory.rs"));
-include!(concat!(env!("OUT_DIR"), "/SushiswapV2Pair.rs"));
-include!(concat!(env!("OUT_DIR"), "/SushiswapV2Router02.rs"));
+include!(concat!(env!("OUT_DIR"), "/IUniswapLikeRouter.rs"));
 include!(concat!(env!("OUT_DIR"), "/SushiswapV2Factory.rs"));
-include!(concat!(env!("OUT_DIR"), "/GPv2Settlement.rs"));
-include!(concat!(env!("OUT_DIR"), "/GPv2AllowListAuthentication.rs"));
+include!(concat!(env!("OUT_DIR"), "/SushiswapV2Router02.rs"));
+include!(concat!(env!("OUT_DIR"), "/UniswapV2Factory.rs"));
+include!(concat!(env!("OUT_DIR"), "/UniswapV2Router02.rs"));
 include!(concat!(env!("OUT_DIR"), "/WETH9.rs"));
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ethcontract::{
+        futures::future::{self, FutureExt as _, Ready},
+        json::json,
+        jsonrpc::{Call, Id, MethodCall, Params, Value},
+        web3::{error::Result as Web3Result, RequestId, Transport, Web3},
+    };
+
+    #[derive(Debug, Clone)]
+    struct ChainIdTransport(u64);
+
+    impl Transport for ChainIdTransport {
+        type Out = Ready<Web3Result<Value>>;
+
+        fn prepare(&self, method: &str, params: Vec<Value>) -> (RequestId, Call) {
+            assert_eq!(method, "net_version");
+            assert_eq!(params.len(), 0);
+            (
+                0,
+                MethodCall {
+                    jsonrpc: None,
+                    method: method.to_string(),
+                    params: Params::Array(params),
+                    id: Id::Num(0),
+                }
+                .into(),
+            )
+        }
+
+        fn send(&self, _id: RequestId, _request: Call) -> Self::Out {
+            future::ready(Ok(json!(format!("{}", self.0))))
+        }
+    }
+
+    #[test]
+    fn deployment_addresses() {
```

Comment:
nice, I wonder how we can encourage people to add tests here whenever a new contract is added.

Reply:
Maybe we could somehow generate these lines from a constant list

```
include!(concat!(env!("OUT_DIR"), "/WETH9.rs"));
```

and include in this test and assertion each element from the list was verified here...

Reply:
#646